tests/PregWrapperTest.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests;
+
+use PhpCsFixer\PregWrapper;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ *
+ * @covers \PhpCsFixer\PregWrapper
+ *
+ * @internal
+ */
+final class PregWrapperTest extends TestCase
+{
+    public function testMatchFailing()
+    {
+        $this->setExpectedException(
+            'RuntimeException',
+            'Error occurred when calling preg_match.'
+        );
+
+        PregWrapper::match('/(?:\D+|<\d+>)*[!?]/', 'foobar foobar foobar', $matches);
+    }
+
+    /**
+     * @param string $pattern
+     * @param string $subject
+     *
+     * @dataProvider provideCommonCases
+     */
+    public function testMatch($pattern, $subject)
+    {
+        $expectedResult = preg_match($pattern, $subject, $expectedMatches);
+        $actualResult = PregWrapper::match($pattern, $subject, $actualMatches);
```

Comment:
Which test is proving that `Preg::*` works when `preg_*` fails?

Reply:
I've added couple.